Docker error: CODE_MOVED for unknown code block?

Running `wrangler dev --port=8000` inside Docker with:

```
/* eslint-disable */
import { Router } from 'itty-router'

const router = Router()
const routes =  JSON.parse(ROUTES)
const proxyTo = hostname => async request => {

  const url = new URL(request.url);
  const forwardedHost = url.hostname;
  url.hostname = hostname;

  const req = new Request(url, request);
  req.headers.append('X-Forwarded-Host', forwardedHost);

  return fetch(req);
}

routes.forEach(route => {
  router.all(route, proxyTo(FRONTEND_HOST));
});

router.all('/api/*', proxyTo(BACKEND_HOST))
router.all('/*', proxyTo(BACKEND_HOST))

addEventListener('fetch', (e) => {
  e.respondWith(router.handle(e.request))
})
```

Everything works fine if I run this outside of Docker, but if I put everything in Docker, when I want to do the proxy, I get:
```
worker-router-worker-1  | ✘ [ERROR] workerd/jsg/setup.c++:481: error: CODE_MOVED for unknown code block?
worker-router-worker-1  |
worker-router-worker-1  |   workerd/jsg/setup.c++:481: error: CODE_MOVED for unknown code block?
worker-router-worker-1  |   workerd/jsg/setup.c++:481: error: CODE_MOVED for unknown code block?
worker-router-worker-1  |   workerd/jsg/setup.c++:481: error: CODE_MOVED for unknown code block?
worker-router-worker-1  |

worker-router-worker-1  | ✘ [ERROR] workerd/jsg/setup.c++:481: error: CODE_MOVED for unknown code block?
worker-router-worker-1  |
worker-router-worker-1  |   workerd/jsg/setup.c++:481: error: CODE_MOVED for unknown code block?
worker-router-worker-1  |

worker-router-worker-1  | ✘ [ERROR] workerd/server/server.c++:2838: error: Uncaught exception: kj/compat/http.c++:1725: failed: expected headerBuffer.size() < MAX_BUFFER [131072 < 131072]; request headers too large
worker-router-worker-1  |
worker-router-worker-1  |   stack: /usr/local/lib/node_modules/wrangler/node_modules/@cloudflare/workerd-linux-arm64/bin/workerd@265008c /usr/local/lib/node_modules/wrangler/node_modules/@cloudflare/workerd-linux-arm64/bin/workerd@2652620 /usr/local/lib/node_modules/wrangler/node_modules/@cloudflare/workerd-linux-arm64/bin/workerd@2656798 /usr/local/lib/node_modules/wrangler/node_modules/@cloudflare/workerd-linux-arm64/bin/workerd@2622b74
worker-router-worker-1  |
[mf:err] Error: Network connection lost.
[mf:inf] GET /terms 500 Internal Server Error (88ms)
```
Individual URL for the host (eg. front-end) works fine from Docker.
Not exactly sure what the `headerBuffer.size()` and where that could be? Could this due to some host settings within Docker?

--- a/src/workerd/jsg/setup.c++
+++ b/src/workerd/jsg/setup.c++
@@ -504,8 +504,9 @@ void IsolateBase::jitCodeEvent(const v8::JitCodeEvent* event) noexcept {
       } else {
         // TODO(someday): Figure out why this triggers. As of v8 10.3 it actually happens in one
         //   of our tests. This API is very undocumented, though, so I'm not sure what I should do.
-        //   Change this back to DEBUG_FAIL_PROD_LOG once debugged.
-        KJ_LOG(ERROR, "CODE_MOVED for unknown code block?");
+        //   Change this back to DEBUG_FAIL_PROD_LOG once debugged. (It was reduced to INFO logging
+        //   to avoid bothering users of workerd.)
+        KJ_LOG(INFO, "CODE_MOVED for unknown code block?");
       }
       break;
 

